hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -111,15 +113,17 @@ public void release(ResultSet resultSet, Statement statement) {
 		}
 		if ( statement != null ) {
 			final Set<ResultSet> resultSets = xref.get( statement );
-			if ( resultSets == null ) {
-				log.unregisteredStatement();
-			}
-			else {
+			if ( resultSets != null ) {
 				resultSets.remove( resultSet );
 				if ( resultSets.isEmpty() ) {
 					xref.remove( statement );
 				}
 			}
+			else if ( log.isDebugEnabled() ) {
+				// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
+				// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
```

Comment:
As far as I can tell you're changing the behavior here, not just improving performance... But I guess it's for the best.

Reply:
Well, I didn't change anything, it was more to be consistent with the new code above. As I will revert it, I'm gonna revert this part too.